src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceSerializer.cs

```diff
@@ -111,24 +111,11 @@ internal void WriteResourceStartMetadataProperties(IODataJsonLightWriterResource
                 resource,
                 resourceState.IsUndeclared);
 
-            if (typeName != null)
+            if (typeName != null && !String.Equals(typeName, ODataConstants.ContextUriFragmentUntyped, StringComparison.Ordinal))

```

Comment:
If it's empty it will fail the String.Equals test.

---
In reply to: [127327893](https://github.com/OData/odata.net/pull/894#discussion_r127327893) [](ancestors = 127327893)